modules/generative-aws/clients/aws.go

```diff
@@ -0,0 +1,440 @@
+)
+
+var compile, _ = regexp.Compile(`{([\w\s]*?)}`)
+
+func buildBedrockUrl(service, region, model string) string {
+	urlTemplate := "https://%s.%s.amazonaws.com/model/%s/invoke"
+	return fmt.Sprintf(urlTemplate, fmt.Sprintf("%s-runtime", service), region, model)
+}
+
+func buildSagemakerUrl(service, region, endpoint string) string {
+	urlTemplate := "https://runtime.%s.%s.amazonaws.com/endpoints/%s/invocations"
+	return fmt.Sprintf(urlTemplate, service, region, endpoint)
+}
+
+type aws struct {
+	awsAccessKey        string
+	awsSecretKey        string
+	buildBedrockUrlFn   func(service, region, model string) string
+	buildSagemakerUrlFn func(service, region, endpoint string) string
+	httpClient          *http.Client
+	logger              logrus.FieldLogger
+}
+
+func New(awsAccessKey string, awsSecretKey string, timeout time.Duration, logger logrus.FieldLogger) *aws {
+	return &aws{
+		awsAccessKey: awsAccessKey,
+		awsSecretKey: awsSecretKey,
+		httpClient: &http.Client{
+			Timeout: timeout,
+		},
+		buildBedrockUrlFn:   buildBedrockUrl,
+		buildSagemakerUrlFn: buildSagemakerUrl,
+		logger:              logger,
+	}
+}
+
+func (v *aws) GenerateSingleResult(ctx context.Context, textProperties map[string]string, prompt string, cfg moduletools.ClassConfig) (*generativemodels.GenerateResponse, error) {
+	forPrompt, err := v.generateForPrompt(textProperties, prompt)
+	if err != nil {
+		return nil, err
+	}
+	return v.Generate(ctx, cfg, forPrompt)
+}
+
+func (v *aws) GenerateAllResults(ctx context.Context, textProperties []map[string]string, task string, cfg moduletools.ClassConfig) (*generativemodels.GenerateResponse, error) {
+	forTask, err := v.generatePromptForTask(textProperties, task)
+	if err != nil {
+		return nil, err
+	}
+	return v.Generate(ctx, cfg, forTask)
+}
+
+func (v *aws) Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt string) (*generativemodels.GenerateResponse, error) {
+	settings := config.NewClassSettings(cfg)
+	service := settings.Service()
+	region := settings.Region()
+	model := settings.Model()
+	endpoint := settings.Endpoint()
+	targetModel := settings.TargetModel()
+	targetVariant := settings.TargetVariant()
+
+	var body []byte
+	var endpointUrl string
+	var host string
+	var path string
+	var err error
+
+	headers := map[string]string{
+		"accept":       "*/*",
+		"content-type": contentType,
+	}
+
+	if v.isBedrock(service) {
+		endpointUrl = v.buildBedrockUrlFn(service, region, model)
+		host = service + "-runtime" + "." + region + ".amazonaws.com"
+		path = "/model/" + model + "/invoke"
+
+		if v.isAmazonModel(model) {
+			body, err = json.Marshal(bedrockAmazonGenerateRequest{
+				InputText: prompt,
+				//TextGenerationConfig: &textGenerationConfig{
+				//	MaxTokenCount: *settings.MaxTokenCount(),
+				//	StopSequences: settings.StopSequences(),
+				//	Temperature:    *settings.Temperature(),
+				//	TopP:          int(*settings.TopP()),
+				//},
```

Comment:
Can these be removed? Or are they still needed?

Reply:
removed